src/Agent.Plugins/PipelineCache/PipelineCacheServer.cs

```diff
@@ -42,32 +52,175 @@ public class PipelineCacheServer
                     return;
                 }
 
-                //Upload the pipeline artifact.
-                PipelineCacheActionRecord uploadRecord = clientTelemetry.CreateRecord<PipelineCacheActionRecord>((level, uri, type) =>
-                    new PipelineCacheActionRecord(level, uri, type, nameof(dedupManifestClient.PublishAsync), context));
-                PublishResult result = await clientTelemetry.MeasureActionAsync(
-                    record: uploadRecord,
-                    actionAsync: async () =>
+                // Tar the contents of the directory and create a tar, and then upload the tar.
+                /* string tempPath = Path.GetTempPath();
+                IEnumerable<string> absoluteFilesPaths = Directory.GetFiles(path, "*.*", SearchOption.AllDirectories);
+                IEnumerable<string> absolutedirectoriesPaths = Directory.GetDirectories(path, "*.*", SearchOption.AllDirectories);
+
+                Func<IEnumerable<string>, IEnumerable<string>> getRelativePaths = (inputPaths) => {
+                    List<string> outputPaths = new List<string>();
+                    foreach(var x in inputPaths)
                     {
-                        return await dedupManifestClient.PublishAsync(path, cancellationToken);
-                    });
+                        outputPaths.Add(Path.GetRelativePath(path, x));
+                    }
+                    return outputPaths;
+                };
 
-                CreatePipelineCacheArtifactOptions options = new CreatePipelineCacheArtifactOptions
+                IEnumerable<string> relativeFilesPath = getRelativePaths(absoluteFilesPaths);
+                IEnumerable<string> relativeDirectoryPath = getRelativePaths(absolutedirectoriesPaths);
+                var outputFile = Path.Combine(path, Guid.NewGuid().ToString()+"files.txt");
```

Comment:
>path [](start = 46, length = 4)

Don't create files in the given folder

Reply:
tempPath?

---
In reply to: [311652142](https://github.com/microsoft/azure-pipelines-agent/pull/2369#discussion_r311652142) [](ancestors = 311652142)